### PR TITLE
Use FindFirstFileW instead of FindFirstFileExW

### PR DESCRIPTION
**Pull request by AtariDreams**
FindFirstFileW works fine, and searches for name match by default

### CODE REVIEW

File: stl/src/filesys.cpp [hunk 1]
```diff
@@ -111,10 +111,9 @@ _FS_DLL void* __CLRCALL_PURE_OR_CDECL _Open_dir(
         _Wildname.append(L"\\*");
     }
 
-    void* _Handle =
-        FindFirstFileExW(_Wildname.c_str(), FindExInfoStandard, &_Dentry, FindExSearchNameMatch, nullptr, 0);
+    void* _Handle = FindFirstFileW(_Wildname.c_str(), &_Dentry);
     if (_Handle == INVALID_HANDLE_VALUE) { // report failure
-        _Errno = ERROR_BAD_PATHNAME;
+        _Errno = GetLastError();
```
Comment (AlexGuteniev) [inline stl/src/filesys.cpp:116]: This part is a more severe change than the API change, yet not even mentioned in PR description. Possibly it even deserves a separate PR.

**Review by AlexGuteniev — COMMENTED** [review]
I don't think there's enough motivation to change legacy "experimental" filesystem.

**Comment by sylveon** [comment]
I don't understand what we gain from changing this, and as Alex mentions this is the legacy experimental filesystem, not much efforts are being invested into it. Fixes should target the new C++17 filesystem.

**Comment by strega-nil-ms** [comment]
@sylveon is right here; we don't want to make any significant changes to the `experimental` filesystem implementation, and this certainly counts as one. Thank you for the contribution, but I don't think we can take it at this time. We really appreciate you though!
